### DISCUSSION TITLE
[Feature Request] Switch for Forcible charge/ discharge for Tibber / awattar / low energy prices

### DISCUSSION THREAD

**Discussion by lakatusch** [wlcrs/huawei_solar#626] [Ideas]
hi there,

i want to charge my battery at night, when prices under 20 cent per kwh.

For this i need the following switch:

[Energy storage] Forcible charge/discharge
Modbus Adress 47100
Values: 0: Stop 1: Charge 2: Discharge

can you integrate the switch into the integration?

regards
Daniel

**Comment by lakatusch** [comment]
@wlcrs 

PLEASE, take 10 seconds and say what do you mean...

I already did this switch via HA Modbus-Integration, but i run into errors, when HA modbus "pings" und your solar integration run againt the sun2000. And because of i will get Tibber at june this year, i have to switch away from your integration, if the switch isnt realised...

thanks a lot for your time

**Reply by Roving-Ronin** [reply]
@lakatusch 

Have you read the front pge of this integration, namely where it says:

"**Services**

This integration exposes multiple services, allowing you to [actively control the amount of electricity exported to the grid](https://github.com/wlcrs/huawei_solar/wiki/Changing-Active-Power-Control) and [forcibly charge/discharge your battery](https://github.com/wlcrs/huawei_solar/wiki/Force-charge-discharge-battery).

**Services**
To enable these advanced features, you need to select 'Elevate permissions' during the setup of this integration."
___________________________

If you look in the Developer Tool --> Services:

FORCE CHARGE = service: huawei_solar.forcible_charge

![image](https://github.com/wlcrs/huawei_solar/assets/108674933/011076a9-30e2-48e6-a8f0-1e9e7f934d7d)

or alternatively you can use service: huawei_solar.forcible_charge_soc  and specify what SoC level you want the battery to charge to.
___________________________

FORCE DISCHARGE = service: huawei_solar.forcible_discharge  

![image](https://github.com/wlcrs/huawei_solar/assets/108674933/084c0100-4a86-4a01-b991-b5c016f236d4)

iirc 5kWh LUNA = Max recharge is 2500W and 10/15kW LUNA is 5000W

or alternatively you can use service: huawei_solar.forcible_discharge_soc  and specify what SoC level you want the battery to discharge to.


### Note: Please CLOSE this request, given the functionality already exists in the integration.

**Reply by lakatusch** [reply]
Im so sorry...
I have installed the Integration more than one year ago and I dont remind on this Services....

Service call works fine for me, tested it one Minute ago..

Thanks a lot for your reply

**Comment by lakatusch** [comment]
See last comment
